### PR TITLE
[Static Graph] ProjectGraph.GetTargetLists throws on invalid target names

### PR DESCRIPTION
**Pull request by cdmihai**
Just cleanup: clarifying some documentation, validating some inputs, avoiding needless iterations.

### CODE REVIEW

File: src/Build.UnitTests/Graph/ProjectGraph_Tests.cs [hunk 1]
```diff
@@ -764,6 +764,47 @@ public void GetTargetsListsShouldApplyDefaultTargetsOnlyToGraphRoots()
...
+
+                IReadOnlyDictionary<ProjectGraphNode, ImmutableList<string>> targetLists = projectGraph.GetTargetLists(new []{ "Foo" });
+                targetLists.Count.ShouldBe(projectGraph.ProjectNodes.Count);
+                targetLists[GetFirstNodeWithProjectNumber(projectGraph, 1)].ShouldBe(new []{ "Foo" });
+                targetLists[GetFirstNodeWithProjectNumber(projectGraph, 2)].ShouldBeEmpty();
+            }
+        }
+
+        [Fact]
+        public void GetTargetListsReturnsEmptyTargetsForAllNodesWhenDefaultTargetsAreRequestedAndThereAreNoDefaultTargets()
```
Comment (rainersigwald) [inline src/Build.UnitTests/Graph/ProjectGraph_Tests.cs]: Under what circumstance are there no default targets? Doesn't MSBuild always have a default target (if none is explicitly listed, it's the first one defined in the logical project)?

File: src/Build/Graph/ProjectGraph.cs [hunk 2]
```diff
@@ -599,6 +606,19 @@ internal string ToDot(Func<ProjectGraphNode, string> nodeIdProvider)
...
+            void ThrowOnEmptyTargetNames(ICollection<string> targetNames)
+            {
+                if (targetNames == null || targetNames.Count == 0)
+                {
+                    return;
+                }
+
+                if (targetNames.Any(targetName => string.IsNullOrWhiteSpace(targetName)))
+                {
+                    throw new ArgumentException(ResourceUtilities.FormatResourceStringIgnoreCodeAndKeyword("OM_TargetNameNullOrEmpty"));
```
Comment (rainersigwald) [inline src/Build/Graph/ProjectGraph.cs]: Would it be possible to have more context here? Right now it just says there's a null/empty target name somewhere. How do I know where?

File: src/Build.UnitTests/Graph/ProjectGraph_Tests.cs (hunk 1 above)
Comment (cdmihai) [reply src/Build.UnitTests/Graph/ProjectGraph_Tests.cs]: The project file has no targets. Added a comment on it.

File: src/Build/Graph/ProjectGraph.cs [hunk 3]
```diff
@@ -597,6 +606,19 @@ internal string ToDot(Func<ProjectGraphNode, string> nodeIdProvider)
...
+            void ThrowOnEmptyTargetNames(ICollection<string> targetNames)
+            {
+                if (targetNames == null || targetNames.Count == 0)
+                {
+                    return;
+                }
+
+                if (targetNames.Any(targetName => string.IsNullOrWhiteSpace(targetName)))
+                {
+                    throw new ArgumentException(ResourceUtilities.FormatResourceStringIgnoreCodeAndKeyword("OM_TargetNameNullOrEmpty", nameof(GetTargetLists)));
```
Comment (rainersigwald) [inline src/Build/Graph/ProjectGraph.cs]: Is this enough context? I was thinking more along the lines of "what project caused this error?"

Comment (cdmihai) [reply src/Build/Graph/ProjectGraph.cs]: Ah, that's a bit harder to answer. There is no actual project here. These targets are the graph wide targets the whole graph needs to be built with. The closest thing for identification would be the set of graph roots

**Pull request merged**